clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockOutputStream.java

```diff
@@ -259,10 +259,10 @@ private void ensureNotFinished() {
     @Override
     public void close() throws IOException {
         if (!finished) {
+            // basicallly flush the buffer writing the last block
             writeBlock();
+            // write the end block and finish the stream
             writeEndMark();
-            flush();
```

Comment:
It seems that we will not call `out.flush` if user calls `close()` after this patch. Is it OK?